### ISSUE TITLE
Support for Laravel 4

### ISSUE THREAD

**Issue by darsain** [darsain/laravel-console#10]
I haven't actually looked into L4 yet, and dunno when I will (no PHP projects on a horizon for me atm), so if someone can give me a quick write up of how to restructure this for L4, you'd be really helpful :)

Also, some ideas on how to retain support for L3 without creating a new repo?

**Comment by tortuetorche** [comment]
Hi Darsain, Jason has done a great job, see pull request #11

**Comment by darsain** [comment]
I knooooow. I've had a working version for L4 on my local for quite some time, but I just don't know how to address incompatibility with `artisan install` on L3 in a reasonable way. So I don't do anything :)

The problem is that L3 bundles are installed from master branch, which I want to be L4. So pushing the L4 will break L3 `artisan install`, and now I'm just repeating myself :)

**Comment by tortuetorche** [comment]
Why not just naming the Laravel 4 package (and repository) `darsain/console-l4` ?

**Comment by darsain** [comment]
Because versioning into separate repos goes against every fiber of my body.

**Comment by tortuetorche** [comment]
So we need to hack your body to have a console with Laravel 4 ? :stuck_out_tongue_winking_eye:

**Comment by darsain** [comment]
I'll try once more asking about this in #laravel@freenode when I'll have some time. If it will go unanswered (again), I'll just move the L3 version to a separate branch, and provide some alternative commands to install it from there without L3 artisan.

My motivation to keep things backwards compatible can go only that far when Laravel didn't provide a way how to deal with this. Specifying which branch should the bundle install from would solve this problem, but bundles registry has nothing of the sorts.

**Comment by sp0tteh** [comment]
Darsain: Can you post your (re-factored) L4 version into a separate branch for now so we can use it / submit pull requests?

**Comment by darsain** [comment]
The current state is that I just can't fucking catch all errors and send them in JSON back to the client with 200 status code. I've been fighting with this sporadically for the last 4 days, and my patience is over.

I've pushed what I have to master, but I'm telling you the error handling is just broken...

I've created the #13 (what a fitting number) issue to track this. Gonna close the "Port to Laravel 4" issues/PRs now.
